### PR TITLE
feat(web3): add struct/tuple arg support and Uniswap swap/quote actions

### PR DESCRIPTION
**Pull request by eskp**
## Summary

- Add `reshapeArgsForAbi()` utility that restructures flat args arrays into tuple objects for Solidity functions with struct parameters (e.g. Uniswap's `exactInputSingle(ExactInputSingleParams)`)
- Integrate into both `read-contract-core` and `write-contract-core` -- transparent for existing protocols with simple params
- Extend Uniswap protocol with SwapRouter02 and QuoterV2 contracts (4 chains each) and 4 new actions: `swap-exact-input`, `swap-exact-output`, `quote-exact-input`, `quote-exact-output`
- Protocol now has 4 contracts and 11 actions (up from 2 contracts and 7 actions)

## Test plan

- [x] `pnpm vitest run tests/unit/abi-struct-args.test.ts` -- 7 tests pass (empty args, simple passthrough, single tuple, mixed params, edge cases)
- [x] `pnpm vitest run tests/unit/protocol-uniswap.test.ts` -- 14 tests pass (updated counts, contract/action validation)
- [x] `pnpm check` -- lint clean
- [x] `pnpm type-check` -- no TypeScript errors
- [ ] Verify ABI auto-resolution works for SwapRouter02/QuoterV2 on mainnet
- [ ] E2E test with a quote action against a live RPC

### CODE REVIEW

File: specs/abi-struct-args.md [hunk 1]
```diff
@@ -0,0 +1,68 @@
+# Spec: Struct/Tuple Parameter Support for Protocol Step Handlers
```
Comment (eskp) [inline specs/abi-struct-args.md:1]: Hey @joelorzet can you please have a look at this PR when you get a chance next. This spec explains the biggest change. Needed for Uniswap and Morpho

File: keeperhub/plugins/web3/steps/write-contract-core.ts [hunk 2]
```diff
@@ -270,11 +274,19 @@ export async function writeContractCore(
         };
       }
 
+      // Build value override for payable functions (e.g. WETH deposit)
+      const valueOverride = ethValue
+        ? { value: ethers.parseEther(ethValue) }
+        : {};
```
Comment (joelorzet) [inline keeperhub/plugins/web3/steps/write-contract-core.ts]: Here might silently fail if the user inputs some random value like `abc` or `-1`. Would be good to parse it into a try/catch statement with a clear error handling returning something like: 
```{ success: false, error: `Invalid ETH value: "${ethValue}"`}```

**Review by joelorzet — DISMISSED** [review]
Looks good. The reshape approach is clean: pure function, no side effects, transparent for existing protocols. A few things from the review:

- Add a guard for functionAbi.inputs being `undefined` in `reshapeArgsForAbi` (use `?.some()` instead of `.some()`): some ABI fragments from explorers can omit it. `keeperhub/lib/abi-struct-args.ts`
- Consider wrapping `ethers.parseEther(ethValue)` in a **try/catch** with a user-friendly error before it hits the raw ethers throw. `keeperhub/plugins/web3/steps/write-contract-core.ts`

None are blockers. Good to merge after this improvements.

**Comment by eskp** [comment]
> Looks good. The reshape approach is clean: pure function, no side effects, transparent for existing protocols. A few things from the review:
> 
> * Add a guard for functionAbi.inputs being `undefined` in `reshapeArgsForAbi` (use `?.some()` instead of `.some()`): some ABI fragments from explorers can omit it. `keeperhub/lib/abi-struct-args.ts`
> * Consider wrapping `ethers.parseEther(ethValue)` in a **try/catch** with a user-friendly error before it hits the raw ethers throw. `keeperhub/plugins/web3/steps/write-contract-core.ts`
> 
> None are blockers. Good to merge after this improvements.

Addressed, thanks for your review
